### PR TITLE
CORE-176 - Basic Account info and Metrics

### PR DESCRIPTION
**Pull request by AxonC**

### CODE REVIEW

File: app/Nova/Metrics/TotalDivisionAccounts.php [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    }
+
+    /**
+     * Determine for how many minutes the metric should be cached.
+     *
+     * @return  \DateTimeInterface|\DateInterval|float|int
+     */
+    public function cacheFor()
+    {
+        return now()->addMinutes(120);
```
Comment (CalumTowers) [inline app/Nova/Metrics/TotalDivisionAccounts.php]: Do we need to consider caching this for longer?

File: app/Nova/Metrics/TotalNonDivisionAccounts.php [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    }
+
+    /**
+     * Determine for how many minutes the metric should be cached.
+     *
+     * @return  \DateTimeInterface|\DateInterval|float|int
+     */
+    public function cacheFor()
+    {
+        // return now()->addMinutes(5);
```
Comment (CalumTowers) [inline app/Nova/Metrics/TotalNonDivisionAccounts.php]: We should cache this.

File: app/Nova/Role.php [hunk 3]
```diff
@@ -0,0 +1,92 @@
+<?php
+
+namespace App\Nova;
+
+use Illuminate\Http\Request;
+use Laravel\Nova\Fields\ID;
+use Laravel\Nova\Fields\Text;
+
+class Role extends Resource
```
Comment (CalumTowers) [inline app/Nova/Role.php]: Can we scaffold the Policy for this class if we are implementing it in this PR?

File: app/Nova/Metrics/TotalAccounts.php [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    }
+
+    /**
+     * Determine for how many minutes the metric should be cached.
+     *
+     * @return  \DateTimeInterface|\DateInterval|float|int
+     */
+    public function cacheFor()
+    {
+        return now()->addMinutes(120);
```
[the remark this answers is not in the document]
Comment (AxonC) [reply app/Nova/Metrics/TotalAccounts.php]: Could we cache it for 24 hours in line with our cert updates?

[the remark this answers is not in the document]
Comment (CalumTowers) [reply app/Nova/Metrics/TotalAccounts.php]: We could do but accounts are added throughout the day as people...

a) login for the first time
b) connect to the network (and the NetworkData stuff creates their account)

So I'm easy on how long we cache this for.
The current admin panel only caches it for 60 minutes which I think is quite a short period.

TLDR; 24 hours seems fine

File: app/Nova/Account.php [hunk 5]
```diff
@@ -25,15 +26,18 @@ class Account extends Resource
...
+    }
 
     /**
      * The columns that should be searched.
      *
      * @var array
      */
     public static $search = [
-        'id', 'name', 'email',
+        'id', 'email',
```
[the remark this answers is not in the document]
Comment (AxonC) [reply app/Nova/Account.php]: At the moment, I think we will have to add both name_first & name_last

File: app/Nova/Filters/MembershipState.php [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+
+    /**
+     * Get the filter's available options.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @return array
+     */
+    public function options(Request $request)
+    {
+        return [
```
[the remark this answers is not in the document]
Comment (AxonC) [reply app/Nova/Filters/MembershipState.php]: These functions would only return true or false surely?

[the remark this answers is not in the document]
Comment (CalumTowers) [reply app/Nova/Filters/MembershipState.php]: @AxonC You're right. Remind me to actually think before making comments.

Could we use State and then use `findByCode`?

https://github.com/VATSIM-UK/core/blob/b3fc7febd0d17b18a9226d80f6d6f63a2f144bb4/app/Models/Mship/State.php#L46-L49
